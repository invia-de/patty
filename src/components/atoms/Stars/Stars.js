import React from 'react';
import PropTypes from 'prop-types';

import { Star } from '../Icon/Icon';
import cx from '../../../utils/classnames';

import styles from './stars.module.scss';

/**
 * @author [Roman Semko](mailto:roman.semko-extern@invia.de)
 */
const Stars = ({ value, className }) => {
  const stars = Array.from({ length: Math.floor(value) }).map((x, i) => (
    <Star key={i} />
  ));
  if (value % 1 > 0) {
    stars.push(<Star half key={0.5} />);
  }
  return <span className={cx(styles.stars, className)}>{stars}</span>;
};

Stars.propTypes = {
  /** how many stars to show */
  value: PropTypes.number.isRequired,
  /** additional styling classes */
  className: PropTypes.string
};

export default Stars;
