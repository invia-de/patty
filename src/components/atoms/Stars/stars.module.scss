@import 'theme.scss';

$stars-color: $blue !default;

.stars {
  margin: 0 0 6px;
  display: block;
  color: $blue !important;
  font-size: 15px;
}
