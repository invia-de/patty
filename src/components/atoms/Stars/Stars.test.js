import React from 'react';
import Stars from './Stars';
import { render } from '@testing-library/react';

test('Stars renders', () => {
  const { container } = render(<Stars value={3} />);
  expect(container.firstChild).toMatchSnapshot();
});
