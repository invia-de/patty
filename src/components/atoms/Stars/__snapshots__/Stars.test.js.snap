// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stars renders 1`] = `
<span
  class="stars"
>
  <svg
    aria-hidden="true"
    class="icon "
    viewBox="0 0 512 487"
  >
    <path
      d="M256.014,0l63.971,181.097l192.015,4.909l-152.429,116.86l54.648,184.051l-158.205,-108.83l-158.234,108.83l54.681,-184.051l-152.461,-116.859l192.011,-4.909l64.003,-181.098Z"
    />
  </svg>
  <svg
    aria-hidden="true"
    class="icon "
    viewBox="0 0 512 487"
  >
    <path
      d="M256.014,0l63.971,181.097l192.015,4.909l-152.429,116.86l54.648,184.051l-158.205,-108.83l-158.234,108.83l54.681,-184.051l-152.461,-116.859l192.011,-4.909l64.003,-181.098Z"
    />
  </svg>
  <svg
    aria-hidden="true"
    class="icon "
    viewBox="0 0 512 487"
  >
    <path
      d="M256.014,0l63.971,181.097l192.015,4.909l-152.429,116.86l54.648,184.051l-158.205,-108.83l-158.234,108.83l54.681,-184.051l-152.461,-116.859l192.011,-4.909l64.003,-181.098Z"
    />
  </svg>
</span>
`;
